### PR TITLE
[fix]: [SRM-15914]: Fix increase in metric costs for orchestrator_state_machine_queue_size_above_five_count metric

### PR DESCRIPTION
**Pull request by Varsha-lalwani**
## Describe your changes
[orchestrator_state_machine_queue_size_above_five_count
](https://console.cloud.google.com/monitoring/metrics-explorer;duration=P14D?referrer=search&project=prod-setup-205416&pageState=%7B%22xyChart%22:%7B%22constantLines%22:%5B%5D,%22dataSets%22:%5B%7B%22plotType%22:%22LINE%22,%22targetAxis%22:%22Y1%22,%22timeSeriesFilter%22:%7B%22aggregations%22:%5B%7B%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22groupByFields%22:%5B%22metric.label.%5C%22metric_type%5C%22%22%5D,%22perSeriesAligner%22:%22ALIGN_RATE%22%7D%5D,%22apiSource%22:%22DEFAULT_CLOUD%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22filter%22:%22metric.type%3D%5C%22monitoring.googleapis.com%2Fbilling%2Fbytes_ingested%5C%22%20resource.type%3D%5C%22global%5C%22%20metric.label.%5C%22metric_domain%5C%22%3D%5C%22custom.googleapis.com%5C%22%20metric.label.%5C%22metric_type%5C%22%3Dmonitoring.regex.full_match(%5C%22custom.googleapis.com%2Fopencensus%2Forchestrator_state_machine_queue_size_above_five_count%5C%22)%22,%22groupByFields%22:%5B%22metric.label.%5C%22metric_type%5C%22%22%5D,%22minAlignmentPeriod%22:%2286400s%22,%22perSeriesAligner%22:%22ALIGN_RATE%22%7D%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D%7D) This metric started receieving 4x times data then it used to. This is happening because [SERVICE_GUARD_LOG_ANALYSIS](https://harness.atlassian.net/browse/SRM-15911) and [SERVICE_GUARD_TIME_SERIES](https://harness.atlassian.net/browse/SRM-15393) have issues and their LE tasks have started failing. The service guard iterator still created the task every 5 minutes despite of anything. So tasks are piling up. The analysis orchestrator job pushes data every 5 sec when it picks the orchestrator for execution for the above metric. 

One solution was to disable this metric until this issue is fixed, but everytime such issue will occur we will have to disable it.
And it doesn't help to push the data for same task every 5 sec for same thing. so added a cut off that after data is pushed for a task, it won't be pushed again for 3 hours. 

Also, [state_machine_evaluation_time](https://console.cloud.google.com/monitoring/metrics-explorer;duration=P30D?referrer=search&project=prod-setup-205416&pageState=%7B%22xyChart%22:%7B%22constantLines%22:%5B%5D,%22dataSets%22:%5B%7B%22plotType%22:%22LINE%22,%22targetAxis%22:%22Y1%22,%22timeSeriesFilter%22:%7B%22aggregations%22:%5B%7B%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22groupByFields%22:%5B%22metric.label.%5C%22metric_type%5C%22%22%5D,%22perSeriesAligner%22:%22ALIGN_RATE%22%7D%5D,%22apiSource%22:%22DEFAULT_CLOUD%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22filter%22:%22metric.type%3D%5C%22monitoring.googleapis.com%2Fbilling%2Fbytes_ingested%5C%22%20resource.type%3D%5C%22global%5C%22%20metric.label.%5C%22metric_domain%5C%22%3D%5C%22custom.googleapis.com%5C%22%20metric.label.%5C%22metric_type%5C%22%3D%5C%22custom.googleapis.com%2Fopencensus%2Fstate_machine_evaluation_time%5C%22%22,%22groupByFields%22:%5B%22metric.label.%5C%22metric_type%5C%22%22%5D,%22minAlignmentPeriod%22:%2286400s%22,%22perSeriesAligner%22:%22ALIGN_RATE%22%7D%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D%7D) metric we disabled when we got CCM alert. But it only pushes data when the state machine is SUCCESS or COMPLETED. The alert was because  we had done some fixes [SRM-15692](https://harness.atlassian.net/browse/SRM-15692) and [SRM-15389](https://harness.atlassian.net/browse/SRM-15389) and a lot of new tasks were created.

We will validate the cost in QA and then push in prod.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[SRM-15692]: https://harness.atlassian.net/browse/SRM-15692?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54244)

### CODE REVIEW

**Comment by Varsha-lalwani** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImpl.java [hunk 1]
```diff
@@ -193,8 +197,14 @@ private void orchestrateAtRunningState(AnalysisOrchestrator orchestrator) {
...
-      try (AutoMetricContext ignore = metricContextBuilder.getContext(orchestrator, AnalysisOrchestrator.class)) {
-        metricService.incCounter(CVNGMetricsUtils.ORCHESTRATOR_STATE_MACHINE_QUEUE_COUNT_ABOVE_FIVE);
+      if (orchestrator.getLastQueueSizeMetricIteration() != null
+          && Instant.ofEpochMilli(orchestrator.getLastQueueSizeMetricIteration())
+                 .isBefore(clock.instant().minus(
+                     CUT_OFF_TIME_FOR_ORCHESTRATOR_QUEUE_SIZE_METRIC_IN_HOURS, ChronoUnit.HOURS))) {
+        try (AutoMetricContext ignore = metricContextBuilder.getContext(orchestrator, AnalysisOrchestrator.class)) {
+          metricService.incCounter(CVNGMetricsUtils.ORCHESTRATOR_STATE_MACHINE_QUEUE_COUNT_ABOVE_FIVE);
+        }
+        orchestrator.setLastQueueSizeMetricIteration(clock.millis());
```
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImpl.java:207]: Where is this being set for the first time?

Comment (Varsha-lalwani) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/impl/OrchestrationServiceImpl.java:207]: updated

**Comment by Varsha-lalwani** [comment]
trigger uts

**Comment by Varsha-lalwani** [comment]
trigger uts

**Review by n00bitax — APPROVED** [review]

**Comment by Varsha-lalwani** [comment]
trigger messagecheck

**Pull request merged**